### PR TITLE
chore: update firebase-functions and firestore security rules version

### LINKED ISSUE #15: upgrade firebase-functions to latest version
`functions: package.json indicates an outdated version of firebase-functions. Please upgrade using npm install --save firebase-functions@latest in your functions directory.`

https://github.com/next-web-technology/symbol-social-wallet/actions/runs/4156063009/jobs/7189594157

### PR DESCRIPTION
**Pull request by kobakaku**
close #15

### CODE REVIEW

**Review by YasunoriMATSUOKA — CHANGES_REQUESTED** [review]
I guess we need to upgrade firebase-functions on `functions/package.json`. (not `package.json`)
But, it seems firebase-functions is newly installed in `package.json`.
So I guess we need to uninstall firebase-functions from `package.json` and upgrade firebase-functions on `functions` directory.

**Comment by kobakaku** [comment]
close #16

File: firestore.rules [hunk 1]
```diff
@@ -1,3 +1,4 @@
+rules_version = '2';
 service cloud.firestore {
   match /databases/{database}/documents {
     match /{document=**} {
```
Comment (YasunoriMATSUOKA) [inline firestore.rules:4]: - [ ] Let's remove unnecessary comments.
- [ ] Let's rewrite the conditional expression as simple as `if false`, etc. (`if request.time < timestamp.date(2023, 3, 10)` is not simple.)

**Review by YasunoriMATSUOKA — CHANGES_REQUESTED** [review]
- [ ] I recommend a little fix in `firestore.rules`
- [ ] I concerned about the diff in `package-lock.json`. Can it be reverted back to the same?

File: firestore.rules [hunk 2]
```diff
@@ -1,16 +1,7 @@
 service cloud.firestore {
```
Comment (YasunoriMATSUOKA) [inline firestore.rules:1]: I think we need to add version 2 settings here.

**Review by YasunoriMATSUOKA — CHANGES_REQUESTED** [review]
I think you need to add firestore rules version 2 settings.

**Pull request merged**